libphal/phal_dump.C

```diff
@@ -285,94 +306,129 @@ void writeDumpFile(char* data, size_t len, std::filesystem::path& dumpPath)
 }
 
 void collectSBEDump(uint32_t id, uint32_t failingUnit,
-		    const std::filesystem::path& dumpPath)
+					const std::filesystem::path& dumpPath, const int sbeTypeId)
 {
+	constexpr int ODYSSEY_SBE_DUMP = 0xB;
+	auto forOdySsey = (ODYSSEY_SBE_DUMP == sbeTypeId);
+
 	log(level::INFO,
-	    "Collecting SBE dump: path=%s, id=%d, chip position=%d",
-	    dumpPath.string().c_str(), id, failingUnit);
+		"Collecting SBE dump: path=%s, id=%d, chip position=%d",
+		dumpPath.string().c_str(), id, failingUnit);
 
 	std::stringstream ss;
 	ss << std::setw(8) << std::setfill('0') << id;
 
 	// Filename format
-	// <dumpID>.<nodeNUM>_<procNum>.Sbedata_p10_<HWP name>
+	// <dumpID>.<nodeNUM>_<procNum>.Sbedata_Ody_/Sbedata_p10_<HWP name>
 	std::string baseFilename =
-	    ss.str() + ".0_" + std::to_string(failingUnit) + "_SbeData_p10_";
+		ss.str() + ".0_" + std::to_string(failingUnit) + (forOdySsey ? "_SbeData_Ody_" : "_SbeData_p10_");
 
 	// Execute pre-collection and get proc corresponding to failing unit
-	struct pdbg_target* proc = preCollection(failingUnit, dumpPath);
+	struct pdbg_target* target = preCollection(failingUnit, dumpPath, sbeTypeId);
 
 	// Get pib for the proc
-	struct pdbg_target* pib = getPibTarget(proc);
+	struct pdbg_target* pib = nullptr;
```

Comment:
Please make this more table driven than these many if else in the code.. 
keep a common struct, and assign the names of correct procedure names for each type of HWP.
for example

struct funcs { extractrc};
proc.extractrc = p10_extract_rc;
like that